### DISCUSSION TITLE
Compact syntax for standard UML class connection/arrow types

### DISCUSSION THREAD

**Discussion by mhlr** [terrastruct/d2#1349] [Q&A]
It is possible to specify the necessary aroow types using general property syntax on connections, but this becomes unusable for largish UML class diagrams.

Is there a concise way to specify composition, aggregation, dependency, inheritance and implementation,
without wrting out all the JSON dicts?

Similar to what Mermaid & palnUML do
 - argregation = o--
 - composition = *--
 - dependency = <--
 - inheritance = <|--
 - etc

Maybe something like this already exists but I am not finding it in the docs

**Comment by alixander** [comment]
We won't do those compact syntaxes, see https://d2lang.com/tour/design#readability--prototyping-speed

But you can do these arrowheads. https://d2lang.com/tour/connections/#arrowheads
